File: reb_4_1/reb_4_1_calculations.py
import pandas as pd

# Given
rho_bed = 155.0 * 0.0283 # lbm/m^3
k0_NH3 = 1.54e15 # kmol NH3 / m^3 / h

# Calculate k0 for generalized rate normalized per pound
k0 = k0_NH3/2/rho_bed

# Display the result
print(' ')
print('k0 (kmol/lbm/h):','%.3g' % k0)
print(' ')

# Save the result to a .csv file
data = [['k0', '%.3g' % k0]]
result = pd.DataFrame(data, columns=['item','value'])
result.to_csv("./reb_4_1/reb_4_1_Python_results.csv", index=False)

File: reb_4_1/reb_4_1_calculations.m
function reb_4_1_calculations
    % Given
    rho_bed = 155 * 0.0283; % lbm/m^3
    k0_NH3 = 1.54E15; % kmol NH3/ m^3 /h

    % Calculate k0 for reaction rate expression normalized per pound
    k0 = k0_NH3/2/rho_bed;

    % Display the results
    disp(' ')
    disp(['k0: ', num2str(k0,3),' kmol/lbm/h'])

    % Save the result to a .csv file
    results_file ="reb_4_1_Matlab_results.csv";
    item = "k0";
    value = round(k0,3,"significant")
    results_table = table(item,value);
    writetable(results_table,results_file);
end